### PR TITLE
Ll insertions

### PR DESCRIPTION
**Pull request by mattburger**

### CODE REVIEW

**Comment by mattburger** [comment]
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

File: code401Challenges/src/main/java/code401Challenges/LinkedList.java [hunk 1]
```diff
@@ -4,6 +4,89 @@
...
+    //add node to end of list
+    protected void append(Object data){
+        Node newNode = new Node(data);
+        newNode.next = null;
+        Node lastNode;
+
+        //if head is null, make head the new value
+        if(this.head == null){
+            this.head = newNode;
+
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/LinkedList.java]: Extra empty line

File: code401Challenges/src/main/java/code401Challenges/LinkedList.java [hunk 2]
```diff
@@ -4,6 +4,89 @@
...
+                    Node tmp = findVal.next;
+                    findVal.next = newNode;
+                    findVal.next.next = tmp;
+                }
+            }
+            catch (NullPointerException err){
+                throw new IllegalArgumentException("Bad target value");
+            }
+            //}
+//            else{
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/LinkedList.java]: Corpse Code!

File: code401Challenges/src/main/java/code401Challenges/LinkedList.java [hunk 3]
```diff
@@ -4,6 +4,89 @@
...
+    //insertBefore
+    protected void insertBefore(Object value, Object newValue){
+        if(this.head == null){
+            this.head = new Node(newValue);
+            this.head.next = null;
+        }
+        else{
+            Node newNode = new Node(newValue);
+            Node findVal = this.head;
+            try {
```
Comment (evanslaton) [inline code401Challenges/src/main/java/code401Challenges/LinkedList.java]: I don't think the try / catch is necessary
